docs/content/cache.md

```diff
@@ -126,33 +129,56 @@ and cloud providers, the cache remote can split multiple requests to the
 cloud provider for smaller file chunks and combines them together locally
 where they can be available almost immediately before the reader usually
 needs them.
+
 This is similar to buffering when media files are played online. Rclone
 will stay around the current marker but always try its best to stay ahead
 and prepare the data before.
 
-#### Warmup mode ####
+#### Plex Integration ####
+
+There is a direct integration with Plex which allows cache to detect during reading
+if the file is in playback or not. This helps cache to adapt how it queries
+the cloud provider depending on what is needed for.
+
+Scans will have a minimum amount of workers (1) while in a confirmed playback cache
+will deploy the configured number of workers.
 
-A negative side of running multiple requests on the cloud provider is
-that you can easily reach a limit on how many requests or how much data
-you can download from a cloud provider in a window of time.
-For this reason, a warmup mode is a state where `cache` changes its settings
-to talk less with the cloud provider.
+This integration opens the doorway to additional performance improvements
+which will be explored in the near future.
 
-To prevent a ban or a similar action from the cloud provider, `cache` will
-keep track of all the open files and during intensive times when it passes
-a configured threshold, it will change its settings to a warmup mode.
+**Note:** If Plex options are not configured, `cache` will function with its
+configured options without adapting any of its settings.
 
-It can also be disabled during single file streaming if `cache` sees that we're
-reading the file in sequence and can go load its parts in advance.
+How to enable? Run `rclone config` and add all the Plex options (endpoint, username
+and password) in your remote and it will be automatically enabled.
 
 Affected settings:
-- `cache-chunk-no-memory`: _disabled_
-- `cache-workers`: _1_
-- file chunks will now be cached using `cache-warm-up-age` as a duration instead of the
-regular `cache-chunk-age`
+- `cache-workers`: _Configured value_ during confirmed playback or _1_ all the other times
 
 ### Known issues ###
 
+#### cache and optimized remotes ####
```

Comment:
@ncw I would be interested what's your opinion about this.

It seems that the 2 http options are random as I have encountered bans and while looking at the request headers they are sent with Seek instead of Range even for `drive` which is definitely the least desirable thing.

For now, I'm looking at what I'm wrapping and searched the code for remotes that directly mention the headers but I believe that's not the best way to look at this isn't it?

Sure, an optional feature or something would be better but for now I would like to get this in as it is as it has (in my opinion) way too many fixes and improvements to wait.

Reply:
One thing which would help a lot would be to put the RangeOption last in the slice when you call Open.

In fs/options.go in [OpenOptionAddHTTPHeaders](https://github.com/ncw/rclone/blob/master/fs/options.go#L120-L127)  the headers are added to the request in order, but it uses `headers.Set` which means the last one will win.

This will work for drive for definite and any of the rest module based remotes.  It works for acd, dropbox.  I think it will work for nearly all the remotes actually.

Reply:
Thanks a bunch @ncw This seems to work great and much more cleaner too